specs/graph-protocol-hybrid-network/read-interface/README.md

```diff
@@ -1,357 +1,273 @@
+```graphql
+type Musician @entity {
+    id: ID!
+    name: String!
+    mainBand: Band
+    bands: [Band!]!
+    writtenSongs: [Song]! @derivedFrom(field: "writtenBy")
+}
 
-This database model is referenced as `entitydb` in Index Records.
+type Band @entity {
+    id: ID!
+    name: String!
+    members: [Musician!]! @derivedFrom(field: "bands")
+    originalSongs: [Song!]!
+}
 
+type Song @entity {
+    id: ID!
+    title: String!
+    writtenBy: Musician!
+    band: Band @derivedFrom(field: "originalSongs")
+}
+```
 
-##### Example - Entities Stored as Key-Value Pairs
-| Key | Value |
-| --- | ----- |
-| `user:1` | `{ id: "user:1", user: "Alice", age: 17 }` |
-| `user:2` | `{ id: "user:2", user: "Bob", age: 47 }` |
+Queries against such a subgraph can make use of the `musicians`, `bands`, and `songs` table. The ID's of the songs a musician has written can be queried with
+```sql
+    select s.id
+    from musicians as m
+    inner join songs as s on s.written_by = m.id
+    where m.name = 'John Coltrane'
+```
 
-#### Partitions
-Partitions define the subset of the data that is covered by the index.
+A GraphQL query that lists musicians, their main bands, and the songs they have written
+```graphql
+    musicians { id mainBand { id } writtenSongs { id } }
+```
 
-Possible values of `<partition>` in an Index Record:
- - none - Includes all entities in the dataset. Default partition, `partition` key should be ommitted in IndexRecord.
- - `"<entityType>"` - Includes entities of the type specified by `entityType`. The entity type name is case-sensitive.
- - `"<interface>"` - Includes entities that implement the provided interface. The interface name is case-sensitive.
+would correspond to this SQL query
+```sql
+    select m.id,
+           map(main_band) as mainBand,
+           collect(songs) as songs over m.id
+        from musicians m
+             left outer join (select id from bands) as main_band
+                             on (m.main_band = main_band.id)
+             left outer join (select id from songs) as songs
+                             on (songs.written_by = m.id)
+```
 
-### Index Types
+The result of the query will look like
+```js
+{
+  "data": [
+    { "id": "coltrane",
+      "mainBand": { "id": "classic_quartet" },
+      "songs": [{"id": "resolution"}, {"id": "psalm"}]
+    },
+    ...
+  ]
+  // remaining response fields omitted
+}
+```
 
-#### Entity DB Indexes
+## Attestation
 
-##### Dictionary
-The entity dictionary supports simple key-value lookups of entities by their entity type and ID, in constant time.
+The attestation contains a verifiable assurance that the data returned for a query is a correct response for that query.
 
-###### Name
-`dictionary`
+The details of how to compute an attestation based on the query, the data retrieved, and the block hashes at which the data was retrieved still need to be defined.
 
-###### Database Model
-`entitydb`
+## Cost model
```

Comment:
Let's link to a tracking issue here as well, which maybe outlines some of the options mentioned here.

Reply:
Tracked here: https://github.com/graphprotocol/graph-node/issues/1098